dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -105,6 +113,152 @@ public void testWorkflowLaunchOrNotLaunchBasedOnCredentials() throws IOException
+        Workflow refresh = workflowApi.refresh(workflow.getId());
+        workflowApi.publish(refresh.getId(), SwaggerUtility.createPublishRequest(true));
+
+        Client.main(
+            new String[] { "--config", ResourceHelpers.resourceFilePath("config_file2.txt"), "workflow", "convert", "entry2json", "--entry",
+                toolpath + ":Dockstore_Testing", "--script" });
+        List<String> stringList = new ArrayList<>();
+        stringList.add("\"SmartSeq2SingleCell.gtf_file\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.genome_ref_fasta\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.rrna_intervals\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.fastq2\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.hisat2_ref_index\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.hisat2_ref_trans_name\": \"String\"");
+        stringList.add("\"SmartSeq2SingleCell.stranded\": \"String\"");
+        stringList.add("\"SmartSeq2SingleCell.sample_name\": \"String\"");
+        stringList.add("\"SmartSeq2SingleCell.output_name\": \"String\"");
+        stringList.add("\"SmartSeq2SingleCell.fastq1\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.hisat2_ref_trans_index\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.hisat2_ref_name\": \"String\"");
+        stringList.add("\"SmartSeq2SingleCell.rsem_ref_index\": \"File\"");
+        stringList.add("\"SmartSeq2SingleCell.gene_ref_flat\": \"File\"");
+        stringList.forEach(string -> Assert.assertTrue(systemOutRule.getLog().contains(string)));
+    }
+
+    /**
+     * This tests that you are able to download zip files for versions of a workflow
+     */
+    @Test
+    public void downloadZipFile() throws IOException {
+        String toolpath = SourceControl.GITHUB.toString() + "/DockstoreTestUser2/md5sum-checker/test";
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+
+        // Register and refresh workflow
+        Workflow workflow = workflowApi
+            .manualRegister(SourceControl.GITHUB.getFriendlyName(), "DockstoreTestUser2/md5sum-checker", "/md5sum/md5sum-workflow.cwl",
+                "test", "cwl", null);
+        Workflow refresh = workflowApi.refresh(workflow.getId());
+        Long workflowId = refresh.getId();
+        WorkflowVersion workflowVersion = refresh.getWorkflowVersions().get(0);
+        Long versionId = workflowVersion.getId();
+
+        // Download unpublished workflow version
+        workflowApi.getWorkflowZip(workflowId, versionId);
+        byte[] arbitraryURL = SwaggerUtility.getArbitraryURL("/workflows/" + workflowId + "/zip/" + versionId, new GenericType<byte[]>() {
+        }, webClient);
+        Path write = Files.write(File.createTempFile("temp", "zip").toPath(), arbitraryURL);
```

Comment:
You may want to extract tempFile to a variable and call deleteOnExit
```
File f = File.createTempFile("temp", "zip");
f.deleteOnExit();
...
```

It won't matter on CI, but they will start accumulating in local builds.